src/pomp_loop.c

```diff
@@ -534,7 +537,8 @@ int pomp_loop_idle_add(struct pomp_loop *loop, pomp_idle_cb_t cb,
 	newentry->cb = cb;
 	newentry->userdata = userdata;
 
-	list_push(&loop->idle_entries, &newentry->node);
+	newentry->next = loop->idle_entries;
```

Comment:
This changes the behaviour. Entries where stored at the end, and processed in order. With this it will be called in reverse order and might break some expectation (maybe not explicitly stated in API but one might expect functions added to be called in the same order)